src/app/events/utils/events.ts

```diff
@@ -26,8 +26,7 @@ export function getEventState(course: Course): Option<EventStateWithLabel> {
     courseStatus.HasTestGrading === false
   ) {
     if (
-      courseStatus.EvaluationUntil &&
-      courseStatus.EvaluationUntil >= new Date()
+      courseStatus.EvaluationUntil
```

Comment:
@schefbi Ich verstehe aus dem Beschrieb in der Issue das Problem noch nicht. Geht es um das Feld `EvaluationUntil` oder um `HasEvaluationStarted` oder um ganz was anderes? Was ist das eigentliche Ziel?

Falls das Ziel ist, dass man bis und mit Ende des Tages bewerten kann, dann wäre die Lösung, dass wir statt Datum mit Zeit nur das Datum (d.h. den Tag) berücksichtigen sollen, also sowas:

```
if (
  courseStatus.EvaluationUntil &&
  startOfDay(courseStatus.EvaluationUntil) >= startOfDay(new Date())
) {
```

Beim Beispiel aus der Issue, wo das `EvaluationUntil` den Wert `2022-01-01T00:00:00` hat, könnte man dann bis um `2022-01-01T23:59:59` (Ende des Tages) bewerten...

Reply:
@hupf  `EvaluationUntil` ist nicht für die Steuerung der Noteneigabe nötig, nur noch für das Label > Bewertung bis oder Zwischenbewertung

In `HasEvaluationStarted` wird das Datum `EvaluationUntil` immer eingeschlossen.

Deshalb müssen wir nur noch prüfen ob EvaluationUntil === null  "Zwischenbewertung " oder eben nicht null "Bewertung bis".

Reply:
Rein mit den mir verfügbaren Infos und den statischen StatusIds im Code ist es mir nicht möglich Sinn/Unsinn der fachlichen Logik zu beurteilen. So wie du das beschreibst scheint die Änderung aber mit dem zu übereinstimmen wie es nun implementiert ist.

Das `EvaluationUntil` mit Datumsvergleich wird noch weiter unten im `canSetFinalGrade` verwendet, welches für die Anzeige es Buttons «Mittelwert als Note setzen» und die Anzeige des Links «Weiter zur Bewertungsübersicht» verwendet wird. Müsste man diese Logik nicht auch noch anpassen?

Habe noch einen Commit gepushed bei dem ich die Logik noch etwas überarbeitet und vereinfacht habe: https://github.com/bkd-mba-fbi/webapp-schulverwaltung/pull/477/commits/9af6875f75662bc5ba2d42f1c931917c8dc94ceb

Wäre gut, wenn du da noch kurz reinschauen könntest. Was mich etwas irritiert ist, dass `HasEvaluationStarted` etc. optionale Flags sind (`boolean | undefined`), die man dann im Code immer mit `===` vergleichen muss. Es wäre wesentlich schöner, wenn die einfach `boolean` wären mit einem Defaultwert...

Reply:
Habe mit Hilfe von Franziska's Infos für die statischen StatusIds Konstanten definiert...